leshan-server-core/src/main/java/org/eclipse/leshan/server/bootstrap/BootstrapConfig.java

```diff
@@ -94,6 +98,28 @@ public String toString() {
         }
     }
 
+    /** oscore configuration (object 17) */
+    static public class OscoreObject implements Serializable {
+
+        public int objectInstanceId;
+        public byte[] oscoreMasterSecret = new byte[] {};
+        public byte[] oscoreSenderId = new byte[] {};
+        public byte[] oscoreRecipientId = new byte[] {};
+        public String oscoreAeadAlgorithm = "";
+        public String oscoreHmacAlgorithm = "";
+        public byte[] oscoreMasterSalt = new byte[] {};
+        public byte[] oscoreIdContext = new byte[] {};
```

Comment:
Yes, adding the ID Context and having incorrect types are mistakes. For the object instance id I added it to be able to link the OSCORE object to the security object of the bootstrap server. That is what I then check in the ConfigurationChecker:
https://github.com/eclipse/leshan/blob/1893811749c6d1fb07e72dff6cdc29b7330e55ba/leshan-bsserver-demo/src/main/java/org/eclipse/leshan/server/bootstrap/demo/ConfigurationChecker.java#L61

Without that ID I am not sure how I can associate an OSCORE object to the BS security object, and (later) one OSCORE object to the DM security object. Do you have any suggestion regarding this?

Regarding the ID Context, it is my understanding from talking to some people at Ericsson that they wish to add this as an item in the OSCORE object.
Do you think it is a solution for now to keep the ID Context input boxes on the Leshan demo server and Leshan bootstrap server but disabled? Or should I hide them, or delete them completely?

I have now made commit https://github.com/eclipse/leshan/pull/857/commits/eb1f6ffbd7108d0e2113c69b0c96d7daf97860dc with the following updates:

- Disabled the ID Context input on bootstrap and server demo apps
- Corrected types for OSCORE object items that were previously byte arrays
- Made the algorithms in the OSCORE object have the correct type

Reply:
The OSCORE  object instance id is the key in the map. In your case : `o.getKey()`

> Regarding the ID Context, it is my understanding from talking to some people at Ericsson that they wish to add this as an item in the OSCORE object

Meaning changing the LWM2M specification v1.1 ? This is something which could be difficult especially when the version is already released.

> Do you think it is a solution for now to keep the ID Context input boxes on the Leshan demo server and Leshan bootstrap server but disabled? Or should I hide them, or delete them completely?

I would go for deleting it for now waiting this was clarified ?

Reply:
> The OSCORE object instance id is the key in the map. In your case : `o.getKey()`

I see, thanks. I will work on updating things accordingly.

> > Regarding the ID Context, it is my understanding from talking to some people at Ericsson that they wish to add this as an item in the OSCORE object
> 
> Meaning changing the LWM2M specification v1.1 ? This is something which could be difficult especially when the version is already released.

That would be for version 1.2 or later.

> > Do you think it is a solution for now to keep the ID Context input boxes on the Leshan demo server and Leshan bootstrap server but disabled? Or should I hide them, or delete them completely?
> 
> I would go for deleting it for now waiting this was clarified ?

Yes, it would be best to delete then. Is it fine if I wait and do this in a separate PR? The reason is that there is input for ID Context in client demo, bsserver demo and server demo. So deleting that functionality would touch all of them.

Reply:
> Is it fine if I wait and do this in a separate PR?

Yep :)

Reply:
I have now made a commit to delete the separate object instance id I had in the OSCORE object. See commit https://github.com/eclipse/leshan/commit/85cb80c6c3c9ec277805904b8f912df9dd9561e7
So now I assign 0 to the oscoreSecurityMode in the security object for the Bootstrap Server and o.getKey() gives me object instance id 0 for that OSCORE object.

Then I guess 1 can be assigned to the oscoreSecurityMode in the security object for the Device Manager and o.getKey() for this second OSCORE object will be 1. As far as I understand this ID is what is assigned in bsconfigstore.js when I do `newConfig.oscore[i] = bs.oscore;` so `newConfig.oscore[i+j] = dm.oscore;` would give me id 1.

> Yep :)

Okay, sounds good.